test/mix/tasks/compile.make_test.exs

```diff
@@ -138,6 +139,28 @@ defmodule Mix.Tasks.Compile.ElixirMakeTest do
     end
   end
 
+  test "user-defined executable through environment variable" do
+    in_fixture fn ->
+      System.put_env("MAKE", "nonexistentmake")
+      with_project_config [], fn ->
+        assert_raise Mix.Error, ~r/not found in the path/, fn ->
```

Comment:
We may want to test for the name of the non-found executable to be correct:

```elixir
assert_raise Mix.Error, ~r/`nonexistentmake` not found in the path/, fn -> ... end
```

Wdyt? Same goes in the test below.